### ISSUE TITLE
problem moving .ready archive status to .done

### ISSUE THREAD

**Issue by vmihailenco** [wal-e/wal-e#203]
I regularly see this after backup:

```
wal_e.main   CRITICAL MSG: problem moving .ready archive status to .done
        DETAIL: Traceback is: Traceback (most recent call last):
          File "/var/lib/postgresql/venv/local/lib/python2.7/site-packages/wal_e/worker/pg/wal_transfer.py", line 60, in mark_done
            os.rename(ready_metadata, done_metadata)
        OSError: [Errno 2] No such file or directory
```

Can you please advice what can cause such behaviour? What is the easiest fix?

**Comment by fdr** [comment]
First bug of this type I've seen. You may want to add instrumentation as to what paths are in these variables and figure out why it's missing.

All this does is do something like path.join(database_directory, pg_xlog). Perhaps you've configured WAL to be somewhere else? I don't know one way or another if that's possible.

**Comment by vmihailenco** [comment]
Nvm. What actually happens is that wal-e WAL uploader hangs infinitely (without logging anything) and files in pg_xlog folder are not deleted. And I tried to debug that by running wal-e manually from the CLI and got the error I described. Apparently I have to run wal-e from different folder and/or with different args.

I fixed my issue by killing wal-e process. Postgres restarted WAL uploading and soon all WAL files were uploaded.
